### PR TITLE
fix: lowercase email on universal auth sendcode route

### PR DESCRIPTION
**Pull request by danielgent**

### CODE REVIEW

File: src/routes/tsx/routes.tsx [hunk 1]
```diff
@@ -570,6 +570,7 @@ export const loginRoutes = new OpenAPIHono<{ Bindings: Env }>()
       const { state } = ctx.req.valid("query");
       // backwards compatible to maintain pasted code
       const params = ctx.req.valid("form");
+      const postedUsername = params.username.toLowerCase();
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:573]: :thinking: with the current implementation where we pass the email address around in the query string _but_ allow it to be changed in almost every step... we'll need to be careful we always lowercase :see_no_evil:

File: test/integration/flows/code-flow.spec.ts [hunk 2]
```diff
@@ -1048,7 +1050,8 @@ describe("code-flow", () => {
     });
 
     expect(env.data.emails.length).toBe(1);
-    const otp = getOTP(env.data.emails[0]);
+    const { otp, to } = getOTP(env.data.emails[0]);
+    expect(to).toBe("john-doe@example.com");
```
Comment (danielgent) [inline test/integration/flows/code-flow.spec.ts:1054]: :eyes: this is the change

File: test/integration/login/code-flow-liquidjs.spec.ts [hunk 3]
```diff
@@ -540,4 +540,65 @@ describe("Login with code on liquidjs template", () => {
...
+    const location = response.headers.get("location");
+
+    const stateParam = new URLSearchParams(location!.split("?")[1]);
+
+    const query = Object.fromEntries(stateParam.entries());
+
+    const postSendCodeResponse = await oauthClient.u.code.$post({
+      query: { state: query.state },
+      form: {
+        username: "JOHN-DOE@example.com",
```
Comment (danielgent) [inline test/integration/login/code-flow-liquidjs.spec.ts:567]: note below it's always `john-doe@example.com`

File: src/routes/tsx/routes.tsx (hunk 1 above)
Comment (markusahlstrand) [reply src/routes/tsx/routes.tsx:573]: The question is where we should lowercase it? I think there's two options: either in every place it enters the code or in the place where it actually matters (or both..). One option is to to it on the zod-schema? Think that's pretty clean? And probably also in actual login logic to be safe..

**Review by markusahlstrand — APPROVED** [review]
This is ok, but consider if it would be better as a transform on row 557?

File: src/routes/tsx/routes.tsx (hunk 1 above)
Comment (danielgent) [reply src/routes/tsx/routes.tsx:573]: I'll try the Zod schema now as you mentioned this here https://github.com/sesamyab/auth/pull/805#discussion_r1601304336

File: src/routes/tsx/routes.tsx [hunk 4]
```diff
@@ -554,7 +554,7 @@ export const loginRoutes = new OpenAPIHono<{ Bindings: Env }>()
           content: {
             "application/x-www-form-urlencoded": {
               schema: z.object({
-                username: z.string(),
+                username: z.string().transform((u) => u.toLowerCase()),
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:557]: @markusahlstrand I reverted the previous commit where I did this manually :+1: 


This seems the way to me.  *next* I'll change the code on this PR https://github.com/sesamyab/auth/pull/805#discussion_r1601304336 to do a transform

Comment (danielgent) [reply src/routes/tsx/routes.tsx:557]: You wrote `And probably also in actual login logic to be safe` but I would disagree with that. We know this works, we have a test for it.

I'd rather have a million tests than duplicate code once just to be safe :smile:

**Pull request merged**